src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -84,6 +84,28 @@ public sealed class CompletionItem : IComparable<CompletionItem>
         /// </summary>
         internal Document Document { get; set; }
 
+        /// <summary>
```

Comment:
might be better than have InternalProperty and create WellKnownInternalPropertyNames with all these doc comments. so that one doesn't need to do all these plumbings whenever there is new item to add. 

whenever it happens, it looks like a lot of plumbing code here and in test. test also looks like better if there is TestOption or something so that, no plumbing is needed when new item is added.